### PR TITLE
[SYCLomatic] Update dpct.exe's visible properties on Windows platform

### PR DESCRIPTION
**Pull request by delecui**
Signed-off-by: Cui, Dele <dele.cui@intel.com>

### CODE REVIEW

**Comment by delecui** [comment]
E2E test is updated in https://github.com/oneapi-src/SYCLomatic-test/pull/83

**Review by zhiweij1 — APPROVED** [review]

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/cmake/modules/AddClang.cmake [hunk 1]
```diff
@@ -33,12 +33,21 @@ endfunction(clang_tablegen)
...
-      VERSION_MINOR ${CLANG_VERSION_MINOR}
-      VERSION_PATCHLEVEL ${CLANG_VERSION_PATCHLEVEL}
-      VERSION_STRING "${CLANG_VERSION} (${BACKEND_PACKAGE_STRING})"
-      PRODUCT_NAME "clang")
+# SYCLomatic_CUSTOMIZATION begin
+      set_windows_version_resource_properties(${name} ${windows_resource_file}
+        VERSION_MAJOR ${CLANG_VERSION_MAJOR}
+        VERSION_MINOR ${CLANG_VERSION_MINOR}
+        VERSION_PATCHLEVEL ${CLANG_VERSION_PATCHLEVEL}
+        VERSION_STRING "${CLANG_VERSION}"
```
Comment (zhimingwang36) [inline clang/cmake/modules/AddClang.cmake:41]: Only 41 and 40 are updated.  So only add conditional code control for those two lines.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**